processing/src/main/java/org/apache/carbondata/processing/datatypes/ArrayDataType.java

```diff
@@ -100,13 +102,15 @@ public ArrayDataType(String name, String parentName, String columnId) {
    * @param parentName
    * @param columnId
    * @param isDictionary
+   * @param nullFormat
    */
   public ArrayDataType(String name, String parentName, String columnId,
-      Boolean isDictionary) {
+      Boolean isDictionary, String nullFormat) {
```

Comment:
We no need to handle this way. Here, compaction is because of logging bad record error for inserting NULL for previous segments. Bad record should be done for Load/Insert flow only. We can avoid performing Bad record validation during validation by passing a flag from global sort flow. Please revert these changes, as it might induce new issues if actual data is NULL/empty data type

Reply:
done